cli-files/go-cat/cmd/root.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright © 2022 NAME HERE <EMAIL ADDRESS>
+*/
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	ErrNoPath = errors.New("Please provide a file name or path to a file")
+	ErrDir    = errors.New("Please provide a file name or path to a file, cannot read a directory")
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "go-cat",
+	Short: "go-cat allows you to see contents of a file",
+	Long:  `go-cat allows you to see contents of a file, just like the cat command`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) == 0 {
+			return ErrNoPath
+		}
+
+		path := args[0]
+
+		argStat, err := os.Stat(path)
+		if err != nil {
+			return err
+		}
+
+		if argStat.IsDir() {
+			return ErrDir
+		}
+
+		data, err := os.ReadFile(args[0])
+
+		if err != nil {
+			return err
+		}
+
+		fmt.Println(string(data))
```

Comment:
Makes sense, thanks. How do we handle it in case of reading file line by line? Can we use `Println` in such case because it's going to print line by line - or do we need to use `Print` for the last line? I understand we won't always need to print what we read - but curious